content/v2.0/reference/storage-engine.md

```diff
@@ -0,0 +1,174 @@
+fsync() is expensive _in terms of time_ but guarantees your data is safe on disk.
+**Important** to batch your points (send in ~2000 points at a time), to fsync() less frequently.
+
+<!-- On read side: -->
+When the storage engine restarts,
+<!-- (if we've pulled the plug, say) -->
+open WAL file and read it back into the in-memory database.
+Answer requests to the /read endpoint.
+
+<!-- ===== V1 material -->
+<!-- TODO is this still true? -->
+<!-- On the file system, the WAL is made up of sequentially numbered files (`_000001.wal`). -->
+<!-- The file numbers are monotonically increasing and referred to as WAL segments. -->
+<!-- When a segment reaches 10MB in size, it is closed and a new one is opened. Each WAL segment stores multiple compressed blocks of writes and deletes. -->
+<!-- Each entry in the WAL follows a [TLV standard](https://en.wikipedia.org/wiki/Type-length-value) with a single byte representing the type of entry (write or delete), a 4 byte `uint32` for the length of the compressed block, and then the compressed block. -->
+
+{{% note%}}
+Once you receive a response to a write request, your data is on disk!
+{{% /note %}}
+
+## Cache
+
+Queries to the storage engine will merge data from the Cache with data from the TSM files.
+Queries execute on a copy of the data that is made from the cache at query processing time.
+This way writes that come in while a query is running won’t affect the result.
+
+The Cache is an in-memory copy of all data points current stored in the WAL.
+The points are organized by the key, which is the measurement, tag set, and unique field.
+Each field is kept as its own time-ordered range.
+The Cache data is not compressed while in memory.
+The cache is recreated on restart by re-reading the WAL files on disk back into memory.
+Deletes sent to the Cache will clear out the given key or the specific time range for the given key.
+
+<!-- It is queried at runtime and merged with the data stored in TSM files. -->
+
+## Time-Structured Merge Tree (TSM)
+
+In order to efficiently compact and store data,
+We group field values grouped by series key, then order field values by time.
+**Time-Structured Merge Tree** (TSM) is our data format.
+TSM files store compressed series data in a columnar format.
+Within a series, we store only differences between values, which is more efficient.
+Column-Oriented storage means we can read by series key and ignore what it doesn't need.
+
+<!-- TERMS -->
+Some terminology:
+
+- a *series key* is defined by measurement, tag key+value, and field key.
+- a *point* is a series key, field value, and timestamp.
+
+After fields are stored safely in TSM files, WAL is truncated...
+<!-- TODO what next? -->
+
+There’s a lot of logic and sophistication in the TSM compaction code.
+However, the high-level goal is quite simple:
+organize values for a series together into long runs to best optimize compression and scanning queries.
```

Comment:
maybe the goal?: To optimize compression and queries, TSM organizes data by series.